js/components/lab-model.js

```diff
@@ -20,8 +20,10 @@ export default class LabModel extends PureComponent {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.model !== prevProps.model) {
-      this.setState({loading: true});
+    if (this.props.model !== prevProps.model && !this.props.liveData) {
+      this.setState({ loading: true });
+    } else {
+      this.setState({ loading: false });
```

Comment:
And you can still reproduce it?  It feels like we're fighting symptoms in a bit random way, not the real cause.

I mean it's not a big deal, but if I saw those lines above, I'd be a bit confused and thinking what's going on here. This logic doesn't make any sense without context of this discussion. But if it's a big investigation, I guess it's not worth it.